pkg/engine/lifecycle_test.go

```diff
@@ -5385,3 +5385,47 @@ func TestExplicitDeleteBeforeReplaceGoSDK(t *testing.T) {
 	}}
 	p.Run(t, snap)
 }
+
+func TestReadResourceGolangLifecycle(t *testing.T) {
+	loaders := []*deploytest.ProviderLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{
+				ReadF: func(urn resource.URN, id resource.ID,
+					inputs, state resource.PropertyMap) (plugin.ReadResult, resource.Status, error) {
+					assert.Equal(t, "someId", id)
```

Comment:
This isn't quite enough validation: there's nothing in here ensuring that `Read` is actually called. Can you add that?

Reply:
Done